### PR TITLE
#41 GitlabLogin (UI button + app config).

### PR DESCRIPTION
**Pull request by criske**
PR for #41

### CODE REVIEW

**Comment by criske** [comment]
@amihaiemil https://github.com/self-xdsd/self-web/blob/55b4ca7c48c7fcbfaeb66336f1add61f30f49478/src/main/resources/public/js/getPublicRepos.js#L10
I think the self-xdsd server should handle public repos request as middleman between client and provider (github, gitlab etc). Right now is hardcoded to "github" but this will break if user is a gitlab one.

**Comment by amihaiemil** [comment]
@criske the endpoint ``/api/users/self`` also returns the ``provider`` as one of the attributes. So that JS function can be addapted easily. Can you leave a puzzle for it? 

Public repos can be read directly from the UI, we don't need a token for that. So it doesn't make sense to do it server-side :D

File: src/main/resources/application.properties [hunk 1]
```diff
@@ -2,4 +2,19 @@ spring.mvc.converters.preferred-json-mapper=jsonb
 
 spring.security.oauth2.client.registration.github.client-id=${gh_client_id}
 spring.security.oauth2.client.registration.github.client-secret=${gh_client_secret}
-spring.security.oauth2.client.registration.github.scope=repo
\ No newline at end of file
+spring.security.oauth2.client.registration.github.scope=repo
+
+spring.security.oauth2.client.registration.gitlab.client-id=${gl_client_id}
+spring.security.oauth2.client.registration.gitlab.client-secret=${gl_client_secret}
```
Comment (amihaiemil) [inline src/main/resources/application.properties:8]: @criske can you make a short PR in the ``self-xdsd/assets`` repository and write these 2 values there? :D

Comment (criske) [reply src/main/resources/application.properties:8]: @amihaiemil sure :)

**Comment by criske** [comment]
@amihaiemil 
> @criske the endpoint `/api/users/self` also returns the `provider` as one of the attributes. So that JS function can be addapted >easily. Can you leave a puzzle for it? :D

The problem is still in there I think, gitlab repos don't have `fork` in the json response, there is an int `forks_count`  for it, though.

**Comment by amihaiemil** [comment]
@criske it's not inportant, we'll make different handlers. The important thing is we can differentiate between the providers :)

**Comment by criske** [comment]
@amihaiemil Or we could pass the whole compare function in the attributes.  js `eval() ` to the rescue :D
But seriously it think it's a good practice to have a little logic as posible on the frontend layer.

**Comment by amihaiemil** [comment]
@criske 
> But seriously I think it's a good practice to have a little logic as posible on the frontend layer.

We don't change the repo json in any way serverside, so in this case it's just a matter of URL, the handler functions would look the same.

Furthermore, reading them serversode would just make it slower :D

**Comment by amihaiemil** [comment]
@rultor merge it please

**Pull request merged**